File: recipes-for-diabetics/src/app/pojo/nutrition.ts
import { Quantity } from "./quantity"

export interface Nutrition {
    quantity:Quantity;
    calories:number;
    totalFat:Quantity|null;
    saturatedFat:Quantity|null;
    transFat:Quantity|null;
    polyUnsaturatedFat:Quantity|null;
    monoUnsaturatedFat:Quantity|null;
    cholestrol:Quantity|null;
    sodium:Quantity|null;
    totalCarbohydrates:Quantity|null;
    dietaryFiber:Quantity|null;
    sugars:Quantity|null;
    protein:Quantity|null;
    calcium:Quantity|null;
    potassium:Quantity|null;
    iron:Quantity|null;
    magnesium:Quantity|null;
    cobalamin:Quantity|null;
    vitaminA:Quantity|null;
    vitaminC:Quantity|null;
    vitaminB6:Quantity|null;
    vitaminD:Quantity|null;
}

File: identity-server/src/main/java/com/avasthi/varahamihir/identityserver/entities/Nutrition.java
package com.avasthi.varahamihir.identityserver.entities;

import com.avasthi.varahamihir.identityserver.pojo.UnitType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.index.Indexed;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public final class Nutrition extends AbstractBase{
    private Quantity quantity;
    private float calories;
    private Quantity totalFat;
    private Quantity saturatedFat;
    private Quantity transFat;
    private Quantity polyUnsaturatedFat;
    private Quantity monoUnsaturatedFat;
    private Quantity cholesterol;
    private Quantity sodium;
    private Quantity totalCarbohydrates;
    private Quantity dietaryFiber;
    private Quantity sugars;
    private Quantity protein;
    private Quantity calcium;
    private Quantity potassium;
    private Quantity iron;
    private Quantity magnesium;
    private Quantity cobalamin;
    private Quantity vitaminA;
    private Quantity vitaminC;
    private Quantity vitaminB6;
    private Quantity vitaminD;
    UnitType unitType;
}
